packages/core/src/index.ts

```diff
@@ -122,28 +128,71 @@ export default class Engine<Name extends string = string> {
 				this.publicParsedRules[name] = rule as RuleNode<Name>
 			}
 		}
+
+		this.publicSituation = {} as Situation<Name>
 	}
 
 	resetCache() {
 		this.cache = emptyCache()
 	}
 
 	setSituation(
-		situation: Partial<Record<Name, PublicodesExpression | ASTNode>> = {},
-		options: { keepPreviousSituation?: boolean } = {},
+		situation: Situation<Name> = {},
+		options: {
+			keepPreviousSituation?: boolean
+			filterSituation?: boolean
+		} = {},
 	) {
 		this.resetCache()
 
 		const keepPreviousSituation = options.keepPreviousSituation ?? false
+		const filterSituation = options.filterSituation ?? false
 
 		Object.keys(situation).forEach((name) => {
+			// We check if the dotteName is a rule of the model
 			if (!(name in this.baseContext.parsedRules)) {
-				throw new PublicodesError(
-					'EvaluationError',
-					`Erreur lors de la mise à jour de la situation : ${name} n'existe pas dans la base de règle.`,
-					{ dottedName: name },
+				const errorMessage = `Erreur lors de la mise à jour de la situation : '${name}' n'existe pas dans la base de règle.`
+
+				if (filterSituation === true) {
+					warning(this.baseContext.logger, errorMessage, { dottedName: name })
+					delete situation[name]
+				} else {
+					throw new PublicodesError('SituationError', errorMessage, {
+						dottedName: name,
+					})
+				}
+			}
+
+			// We check if the value from a mutliple choices question `dottedName`
+			// is defined as a rule `dottedName . value` in the model.
+			// If not, the value in the situation is an old option, that is not an option anymore.
+			// TODO: should use a more robust way to check if the value is a possible answer already implemented elsewhere ?
+			const parsedSituationExpr =
+				typeof situation[name] === 'object' ?
+					situation[name]
+				:	parseExpression(situation[name], name)
+
+			if (
+				parsedSituationExpr?.constant?.type === 'string' &&
+				typeof situation[name] !== 'boolean' &&
+				!(
+					`${name} . ${situation[name]?.replaceAll(/^'|'$/g, '')}` in
+					this.baseContext.parsedRules
```

Comment:
I don't think a more robust way is implemented elsewhere unfortunatly. The mecanism `une possibilité` will definitely be more robust one day, and maybe its api will change slightly in a forecoming version (or a new mecanism will be created).